src/NodeDev.Blazor/Components/ProjectToolbar.razor

```diff
@@ -0,0 +1,81 @@
+﻿@using Microsoft.AspNetCore.Components.Forms
+@using NodeDev.Blazor.Services
+@using NodeDev.Core
+@inject ProjectService ProjectService
+@inject ISnackbar Snackbar
+
+<MudFileUpload T="IBrowserFile" FilesChanged="Open">
+    <ActivatorContent>
+        <MudButton Class="ml-3" Disabled=@Project.IsLiveDebuggingEnabled data-test-id="openProject">Open Project</MudButton>
+    </ActivatorContent>
+</MudFileUpload>
+<MudButton OnClick="NewProject" Class="ml-3" Disabled=@Project.IsLiveDebuggingEnabled data-test-id="newProject">New Project</MudButton>
+<MudButton OnClick="Save" Class="ml-3" data-test-id="Save">Save</MudButton>
+<MudButton OnClick="Add" Class="ml-3">Add node</MudButton>
+<MudButton OnClick="Run" Class="ml-3">Run</MudButton>
+<MudButton OnClick="SwitchLiveDebugging">@(Project.IsLiveDebuggingEnabled ? "Stop Live Debugging" : "Start Live Debugging")</MudButton>
+<MudSpacer />
+<MudIconButton Icon="@Icons.Material.Filled.MoreVert" Color="Color.Inherit" Edge="Edge.End" />
+
+
+@code {
+
+    private Project Project => ProjectService.Project;
+
+    [Parameter]
+    public EventCallback OnProjectChanged { get; set; }
+
+    private IBrowserFile? File;
+
+    private async Task Open(IBrowserFile file)
+    {
+        File = file;
+        if (File != null)
+        {
+            using var stream = file.OpenReadStream();
+            using var reader = new StreamReader(stream);
```

Comment:
There's a "new" IAsyncDisposable that we can use here. Since closing that stream is almost certainly an operation that'll talk with the frontend over the network it's gonna be asynchronous
```suggestion
            await using var stream = file.OpenReadStream();
            await using var reader = new StreamReader(stream);
```